vendor/bat-native-ads/src/bat/ads/internal/ml/pipeline/pipeline_util.cc

```diff
@@ -0,0 +1,220 @@
+/* Copyright (c) 2020 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+#include <map>
+#include <memory>
+#include <string>
+#include <vector>
+
+#include "base/json/json_reader.h"
+#include "bat/ads/internal/ml/data/vector_data.h"
+#include "bat/ads/internal/ml/ml_aliases.h"
+#include "bat/ads/internal/ml/ml_transformation_util.h"
+#include "bat/ads/internal/ml/pipeline/pipeline_info.h"
+#include "bat/ads/internal/ml/pipeline/pipeline_util.h"
+
+namespace ads {
+namespace ml {
+namespace pipeline {
+
+base::Optional<TransformationVector> ParseTransformationsJSON(
+    base::Value* transformations_value) {
+  if (!transformations_value || !transformations_value->is_list()) {
+    return base::nullopt;
+  }
+
+  base::Optional<TransformationVector> transformations = TransformationVector();
```

Comment:
use of `transformations` and `transformations_value` is confusing to read, maybe we could use `transformations` for the list and `transformation_vector` for the vector

Reply:
Done. Completely removed this variable.